src/ConnectionAutomaton.cpp

```diff
@@ -375,12 +376,13 @@ void ConnectionAutomaton::processFrameImpl(
   auto frameType = frameSerializer_->peekFrameType(*frame);
   stats_->frameRead(frameType);
 
+  auto streamIdPtr = frameSerializer_->peekStreamId(*frame);
+
   // TODO(tmont): If a frame is invalid, it will still be tracked. However, we
   // actually want that. We want to keep
   // each side in sync, even if a frame is invalid.
-  resumeCache_->trackReceivedFrame(*frame, frameType);
+  resumeCache_->trackReceivedFrame(*frame, frameType, streamIdPtr);
```

Comment:
Stream id should probably not be Optional, just put this after the `if` below (no stream id is a connection-level error).